lib/wos_queries.rb

```diff
@@ -0,0 +1,279 @@
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [Integer]
+    def query_id(response, response_type)
+      response_return(response, response_type)[:query_id].to_i
+    end
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [Integer]
+    def records_found(response, response_type)
+      response_return(response, response_type)[:records_found].to_i
+    end
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [WosRecords]
+    def records(response, response_type)
+      WosRecords.new(records: response_return(response, response_type)[:records])
+    end
+
+    ###################################################################
+    # Search User Query Helpers
+
+    # Constructs a WoS name query
+    # @param name [String] a CSV name pattern: {last name}, {first_name} [{middle_name} | {middle initial}]
+    def name_query(name)
+      split_name = name.split(',')
+      last_name = split_name[0]
+      first_middle_name = split_name[1]
+      first_name = first_middle_name.split(' ')[0]
+      middle_name = first_middle_name.split(' ')[1]
+      name_query = "#{last_name} #{first_name} OR #{last_name} #{first_name[0]}"
+      name_query += " OR #{last_name} #{first_name[0]}#{middle_name[0]} OR #{last_name} #{first_name} #{middle_name[0]}" unless middle_name.blank?
+      name_query
+    end
+
+    # Search authors from these institutions
+    # @return [Array<String>] institution names
+    def institutions
+      ['Stanford University']
+    end
+
+    ###################################################################
+    # WoS Query Parameters
+
+    # @param uid [String] a WOS UID
+    # @return [Hash] citedReferences parameters
+    def cited_references_params(uid)
+      retrieve_options = [ { key: 'Hot', value: 'On' } ]
+      {
+        databaseId: @database,
+        uid: uid,
+        queryLanguage: QUERY_LANGUAGE,
+        retrieveParameters: retrieve_parameters(options: retrieve_options)
+      }
+    end
+
+    # @param uid [String] a WOS UID
+    # @return [Hash] citingArticles parameters
+    def citing_articles_params(uid)
+      {
+        databaseId: @database,
+        uid: uid,
+        timeSpan: time_span,
+        queryLanguage: QUERY_LANGUAGE,
+        retrieveParameters: retrieve_parameters
+      }
+    end
+
+    # @param uid [String] a WOS UID
+    # @return [Hash] relatedRecords parameters
+    def related_records_params(uid)
+      # The 'WOS' database is the only option for this query
+      {
+        databaseId: 'WOS',
+        uid: uid,
+        timeSpan: time_span,
+        queryLanguage: QUERY_LANGUAGE,
+        retrieveParameters: retrieve_parameters
+      }
+    end
+
+    # @param uids [Array<String>] a list of WOS UIDs
+    # @return [Hash] retrieveById parameters
+    def retrieve_by_id_params(uids)
```

Comment:
A ton of repetition here in the `params` methods.  Would it make sense to just have a `base_params` method (that looks exactly like the body of this one) and then invoke: 
```ruby
base_params.merge(different_key: different_val)
```
in the places where individual key/val pairs need to be added or overridden?  This would put the focus on what is unique to each set of params rather than repeating all the things that are in common.

Reply:
If you look closely at it, there are slight differences in each set of params.  Given that we know the request XML is order-sensitive (right?  why? weird, if so, but yea ...), doesn't this make it difficult to use a Hash.merge strategy with any confidence?  I think we should be glad that savon allows us to use Hash params.  These methods have already been refactored a bit to allow for some existing use-case flexibility; we might discover more as we go.  I'm open to exploring whether or not a Hash.merge strategy might work and will try it out soon.  If it's not a blocker right now, maybe we can move on and adapt as we go?

Reply:
Yes, I acknowledged those slight differences in my comment. My intent is to have a common starting point, then merge/reject from there.  This would make it obvious what each of those differences is, I.E., what is special about *this method*.  

The ordered XML thing does not apply since we are still giving the Savon gem a Hash.  It figures out the order now, and it would figure out the order the same way, since we would be passing it exactly the same param.  This is just about how we build the hash and how much we repeat ourselves.  In general, this codebase has suffered from lack of facility manipulating ruby Hash, so I would like to start repairing that.

Reply:
I'll take another look at this.